### DISCUSSION TITLE
[Bug]在1.10.22lts之后，通过CloudFlare的CDN打开后台无法正常加载，会直接把HTML源码写出来

### DISCUSSION THREAD

**Discussion by 65003118** [1Panel-dev/1Panel#7432] [Q&A]
### Contact Information

stoveonbox@outlook.com

### 1Panel Version

v1.10.22-lts

### Problem Description

[1Panel在1.10.22lts之后，通过CloudFlare的CDN打开后台无法正常加载，会直接把HTML源码写出来，例如
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8" />
    <meta name="robots" content="noindex,nofollow" />
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
    <title>Loading...</title>
    <script type="module" src="/assets/js/main.js"></script>
    <link rel="stylesheet" href="/assets/css/style.css">
</head>
<body>
    <div id="app">
        <style>
            html, body, #app {
                width: 100%;
                height: 100%;
                padding: 0;
                margin: 0;
            }
            .first-loading-wrap {
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                width: 100%;
                height: 100%;
            }
            .first-loading-wrap > h1 {
                font-size: 128px;
            }
            .first-loading-wrap .loading-wrap {
                display: flex;
                align-items: center;
                justify-content: center;
                padding: 98px;
            }
            /* ... rest of the CSS remains unchanged as it's not sensitive ... */
        </style>
        <div class="first-loading-wrap">
            <div class="loading-wrap">
                <span class="dot dot-spin">
                    <i></i><i></i><i></i><i></i>
                </span>
            </div>
        </div>
    </div>
</body>
</html>

```
问题是浏览器没有报错，服务器也没有错误日志，只有再正常不过的访问日志
不通过CLoudFlare，直接SSL是完全没有问题的，所以我推断是CF的问题，但是CF所有的优化关掉和缓存清理掉，也没有用，WAF甚至是空的
但是也不认为是1Panel的问题，毕竟直接SSL完全没问题
1.10.21 lts之前的版本不会有这个问题
总结：
如果你说是CF的问题，那之前的版本没问题
如果你说是1Panel的问题，但是不走CF没问题
如果你说我服务器问题，两台都是这个情况
所以来求助一下开发者，v1.10.21-lts就不会有

### Steps to Reproduce

首先使用CloudFlare CDN解析域名 例如A www.xxx.com 11.11.11.11 在11.11.11.11安装v1.10.22-lts的1Panel,通过OpenResty反代1Panel的端口，例如8443,增加安全入口.例如/safe。此时通过www.xxx.com/safe进行访问1Panel入口会打开一个没有任何渲染的HTML文档。
同理，即使使用1Panel自带的安全域名绑定和SSL绑定不会有什么本质区别

### The expected correct result

应该正常打开1Panel后台入口

注：
在v1.10.22-lts之前的版本，例如v1.10.21-lts完全是正常的，之后的beta开始出现的问题
如果不通过CF的CDN也不会有这个问题，依然可以正常打开1Panel后台入口
但是CF没有WAF也没有优化，并且之前的版本例如v1.10.21-lts就算有WAF和优化也是正常的，也可以打开入口

### Related log output

```shell
OpenResty没有错误日志，CDN处没有拦截日志
```

### Additional Information

![PixPin_2024-12-18_17-56-59](https://github.com/user-attachments/assets/31420615-03ca-4a64-b42b-8d11c67f7678)

**Comment by zhengkunwang223** [comment]
我们在 cloudFlare 测试过 没有出现你这种情况
你先试一下 关闭 安全入口 有没有问题

**Reply by 65003118** [reply]
你说得对，关掉安全入口就正常了，可以进去了

**Reply by 65003118** [reply]
那如果我还是挺喜欢安全入口这个功能我需要额外做什么吗？

**Reply by zhengkunwang223** [reply]
> 那如果我还是挺喜欢安全入口这个功能我需要额外做什么吗？

你写反向代理的时候是不是把安全入口也写进去了？

**Reply by 65003118** [reply]
就是直接通过咱们面板里面默认的OpenResty反代出来的，并且系统设置里的域名绑定和SSL绑定不能自己设置把安全入口写进去

**Reply by zhengkunwang223** [reply]
> 就是直接通过咱们面板里面默认的OpenResty反代出来的，并且系统设置里的域名绑定和SSL绑定不能自己设置把安全入口写进去

没看懂，你贴一下配置吧

**Reply by 65003118** [reply]
```
server {
    listen 443 ssl http2;
    server_name example.com; # 域名信息已脱敏
    index index.php index.html index.htm default.php default.htm default.html;

    # 代理头信息设置
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $server_name;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;

    # 日志设置
    access_log /www/sites/example/log/access.log main; # 路径已脱敏
    error_log /www/sites/example/log/error.log; # 路径已脱敏

    # ACME 挑战
    location ^~ /.well-known/acme-challenge {
        allow all;
        root /usr/share/nginx/html;
    }

    # 包含其他配置文件
    include /www/sites/example/proxy/*.conf; # 路径已脱敏

    # SSL 配置
    ssl_certificate /path/to/ssl/fullchain.pem; # 路径已脱敏
    ssl_certificate_key /path/to/ssl/privkey.pem; # 路径已脱敏
    ssl_protocols TLSv1.3 TLSv1.2;
    ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:...'; # 简化了显示
    ssl_prefer_server_ciphers on;
    ssl_session_cache shared:SSL:10m;
    ssl_session_timeout 10m;

    # 错误页
    error_page 497 https://$host$request_uri;

    # 头信息添加
    proxy_set_header X-Forwarded-Proto https;
    add_header Strict-Transport-Security "max-age=31536000";
}

```

```
location / {
    proxy_pass http://[脱敏IP]:[脱敏端口]; 
    proxy_set_header Host $host; 
    proxy_set_header X-Real-IP $remote_addr; 
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
    proxy_set_header REMOTE-HOST $remote_addr; 
    proxy_set_header Upgrade $http_upgrade; 
    proxy_set_header Connection $http_connection; 
    proxy_set_header X-Forwarded-Proto $scheme; 
    proxy_http_version 1.1; 
    add_header X-Cache $upstream_cache_status; 
    add_header Cache-Control no-cache; 
    proxy_ssl_server_name off; 
    proxy_ssl_name $proxy_host; 
    add_header Strict-Transport-Security "max-age=31536000"; 
}

```

**Reply by zhengkunwang223** [reply]
> ```
> 
> server {
> 
>     listen 443 ssl http2;
> 
>     server_name example.com; # 域名信息已脱敏
> 
>     index index.php index.html index.htm default.php default.htm default.html;
> 
> 
> 
>     # 代理头信息设置
> 
>     proxy_set_header Host $host;
> 
>     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> 
>     proxy_set_header X-Forwarded-Host $server_name;
> 
>     proxy_set_header X-Real-IP $remote_addr;
> 
>     proxy_http_version 1.1;
> 
>     proxy_set_header Upgrade $http_upgrade;
> 
>     proxy_set_header Connection $http_connection;
> 
> 
> 
>     # 日志设置
> 
>     access_log /www/sites/example/log/access.log main; # 路径已脱敏
> 
>     error_log /www/sites/example/log/error.log; # 路径已脱敏
> 
> 
> 
>     # ACME 挑战
> 
>     location ^~ /.well-known/acme-challenge {
> 
>         allow all;
> 
>         root /usr/share/nginx/html;
> 
>     }
> 
> 
> 
>     # 包含其他配置文件
> 
>     include /www/sites/example/proxy/*.conf; # 路径已脱敏
> 
> 
> 
>     # SSL 配置
> 
>     ssl_certificate /path/to/ssl/fullchain.pem; # 路径已脱敏
> 
>     ssl_certificate_key /path/to/ssl/privkey.pem; # 路径已脱敏
> 
>     ssl_protocols TLSv1.3 TLSv1.2;
> 
>     ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:...'; # 简化了显示
> 
>     ssl_prefer_server_ciphers on;
> 
>     ssl_session_cache shared:SSL:10m;
> 
>     ssl_session_timeout 10m;
> 
> 
> 
>     # 错误页
> 
>     error_page 497 https://$host$request_uri;
> 
> 
> 
>     # 头信息添加
> 
>     proxy_set_header X-Forwarded-Proto https;
> 
>     add_header Strict-Transport-Security "max-age=31536000";
> 
> }
> 
> 
> 
> ```
> 
> 
> 
> ```
> 
> location / {
> 
>     proxy_pass http://[脱敏IP]:[脱敏端口]; 
> 
>     proxy_set_header Host $host; 
> 
>     proxy_set_header X-Real-IP $remote_addr; 
> 
>     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
> 
>     proxy_set_header REMOTE-HOST $remote_addr; 
> 
>     proxy_set_header Upgrade $http_upgrade; 
> 
>     proxy_set_header Connection $http_connection; 
> 
>     proxy_set_header X-Forwarded-Proto $scheme; 
> 
>     proxy_http_version 1.1; 
> 
>     add_header X-Cache $upstream_cache_status; 
> 
>     add_header Cache-Control no-cache; 
> 
>     proxy_ssl_server_name off; 
> 
>     proxy_ssl_name $proxy_host; 
> 
>     add_header Strict-Transport-Security "max-age=31536000"; 
> 
> }
> 
> 
> 
> ```

系统设置里面开了域名绑定和SSL?

**Reply by 65003118** [reply]
分别的，使用OpenResty的时候，面板设置里面的安全面板 SSL和域名绑定都是关闭的，如果安全面板 SSL和域名绑定开启，OpenResty里面的页面我会删除，反正就是只留一个

**Reply by zhengkunwang223** [reply]
> 分别的，使用OpenResty的时候，面板设置里面的安全面板 SSL和域名绑定都是关闭的，如果安全面板 SSL和域名绑定开启，OpenResty里面的页面我会删除，反正就是只留一个

两种情况都会出问题是吗？

**Reply by 65003118** [reply]
是的

**Reply by zhengkunwang223** [reply]
> 是的

开启安全入口之后，是直接访问域名有问题，还是访问域名/安全入口有问题呢？

**Reply by 65003118** [reply]
开启以后，如果直接访问会正常提示暂时服务访问，提示需要通过安全入口进入。只是真的通过安全入口的时候，进不去
